src/main/webapp/app/entities/oauth-client/oauth-client-dialog.component.html

```diff
@@ -0,0 +1,75 @@
+<form name="editForm" role="form" novalidate (ngSubmit)="save()" #editForm="ngForm">
+
+    <div class="modal-header">
+        <h4 class="modal-title" id="myOAuthClientLabel" jhiTranslate="managementPortalApp.oauthClient.home.createOrEditLabel">Create or edit an OAuth Client</h4>
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true"
+                (click)="clear()">&times;</button>
+    </div>
+    <div class="modal-body">
+        <jhi-alert-error></jhi-alert-error>
+        <div class="alert alert-danger" *ngIf="protectedClient">
+            <strong>This is a protected client.</strong> You will not be able to make any changes to it.
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="id" jhiTranslate="managementPortalApp.oauthClient.clientId">Client ID</label>
+            <input *ngIf="!newClient" type="text" class="form-control" id="id" name="id" [(ngModel)]="client.clientId" readonly />
+            <input *ngIf="newClient" type="text" class="form-control" id="id" name="id" [(ngModel)]="client.clientId" required />
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.clientSecret">Client Secret</label>
+            <div class="input-group">
+                <input type="text" class="form-control" id="secret" name="clientSecret" [(ngModel)]="client.clientSecret" [disabled]="protectedClient" [required]="newClient" />
+                <span class="input-group-btn"><button class="btn btn-secondary" type="button" (click)="generateRandomSecret()"><span class="fa fa-magic">&nbsp;</span>Randomize</button></span>
+            </div>
+            <small id="scopeHelp" class="form-text text-muted" *ngIf="!newClient">If this field is left empty the secret will not be modified.</small>
+        </div>
+        <div class="form-group">
+            <label class="form-control-label" for="" jhiTranslate="managementPortalApp.oauthClient.scope">Scope</label>
+            <input type="text" class="form-control" id="scope" name="scope" [(ngModel)]="scopeList" [disabled]="protectedClient" required />
```

Comment:
Again trying to be user friendly. Not sure about the complexity. Can we limit the scopes, or provide the available scopes from where you can choose? writing all of this by hand is going to be difficult.